newIDE/app/src/UI/KeyboardShortcuts/index.js

```diff
@@ -143,16 +143,31 @@ export default class KeyboardShortcuts {
   };
 
   _updateModifiersFromEvent = (evt: KeyboardEvent | DragEvent) => {
+    const hasModifierChanged =
+      this._metaPressed !== evt.metaKey ||
+      this._altPressed !== evt.altKey ||
+      this._ctrlPressed !== evt.ctrlKey ||
+      this._shiftPressed !== evt.shiftKey;
+
     this._metaPressed = evt.metaKey;
     this._altPressed = evt.altKey;
     this._ctrlPressed = evt.ctrlKey;
     this._shiftPressed = evt.shiftKey;
+
+    return hasModifierChanged;
   };
 
   _updateSpecialKeysStatus = (evt: KeyboardEvent, isDown: boolean) => {
     if (evt.which === SPACE_KEY) {
+      // Prevent scrolling in Physics2 property editor.
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      const hasStateChanged = this._spacePressed !== isDown;
       this._spacePressed = isDown;
+      return hasStateChanged;
```

Comment:
> I'm not sure returning something from this method is a good idea and the value returned is a bit vague: I tried hitting space and command (2 side by side keys) and the cursor stays with the hand until I hit space again.

I couldn't reproduce the issue. In which view did you had the issue?

> I feel like you should use a new callback in the keyboard shortcut that would be `onSpacePressed(pressed: boolean)`. And if this callback is specified, and only then, you would prevent default and stop the propagation

Is it to be sure this code is only called when the space key is actually watched?
Because it's already the case. This class name is maybe too generic. It's actually only used by views that allow to scroll and zoom on an image/scene.
What to you think about renaming it `GraphicalEditorKeyboardShortcuts`?